### ISSUE TITLE
About code modification & behavior-rich traffic

### ISSUE THREAD

**Issue by jedeka** [angmavrogiannis/B-GAP-Behavior-Guided-Action-Prediction-and-Navigation-for-Autonomous-Driving#3]
Hello,

I would like to ask a little question, did you modify any of the original rl_agents code, especially the DQN agent? Because if you don't mind, I would like to know more details on how or where you shape the reward function. And also, I wonder if you can provide the command for the traffic simulation with richer behavior because it seems to me that the traffic shown in the README.md is much richer than what is simulated with the command.

Anyway, great work from you and your research team! 

Thanks for reading

**Comment by angmavrogiannis** [comment]
Hey, thanks for the question!

No, the DQN agent was not modified*. The reward function used is the same as in the original highway-env simulator. To change the reward values, you can modify the existing rewards in the [highway_env.py](highway_env/envs/highway_env.py) file, which are defined in the default_config function.

You can run the code for training the DQN agent for e.g. 2000 episodes with the following command:

```
python experiments.py evaluate configs/HighwayEnv/env.json configs/HighwayEnv/agents/DQNAgent/dqn.json --train --episodes=2000 --name-from-config
```

You can set the total number of vehicles by changing the value of the "vehicles_count" field in the default_config function, and select the percentage of aggressive traffic by changing the "perc_aggressive" field. A value 0.4 of "perc_aggressive" corresponds to 40% aggressive vehicles, so if the total cars are 50, 0.4 * 50 = 20 cars are aggressive.

\* What was modified in a prior version of the project/paper, was the [models.py](rl-agents/rl_agents/agents/common/models.py) file where I had added code using torch-geometric to define a Graph Convolutional neural network, but it was not producing better results compared to the MLP, so I've since removed this from the repo.

Cheers,
Angelos

**Comment by jedeka** [comment]
Thanks a lot!
